### ISSUE TITLE
No longer working (LGOLED65C9PLA - wos 05.20.15)

### ISSUE THREAD

**Issue by GRX** [RootMyTV/RootMyTV.github.io#38]
Hi,

The exploit is no longer working sadly, the script tells it was a success, then reboots, but after a reboot nothing happened.

**Comment by gotwig** [comment]
Same for me. I just discovered the project yesterday.

Tried on LG C9, latest fw. Run exploit, it rebooted - nothing happened!

**Comment by ivarbanovv** [comment]
Yep me three . Just confirmed that homebrew channel is missing after the 'successful' patch and reboot

**Comment by AlexanderBaransky** [comment]
I confirm, LG NANO86, latest firmware. Exploit is broken :( Nothing changed after reboot

**Comment by theoneandonly-vector** [comment]
maybe someone has a safe method to downgrade?

**Comment by gitgud187** [comment]
Exactly the same for me - exploit says it was successful but no homebrew channel.

**Comment by IceBear2k** [comment]
Doesn't work here either. Exploit successful, TV reboots, no popups or any changes after reboot and no Homebrew Channel.

LG OLED55B97LA
Software Version 05.20.15
webOS TV Version 4.9.5-10 (goldilocks2-gympie)

**Comment by Rangnarok123** [comment]
Just to report that currently this is not working on mine either
Software Version 05.20.15
webOS TV Version 4.9.5-10 (goldilocks2-gympie)

**Comment by iMongui** [comment]
Nope, here is the same, LG C9, latest fw as my other colleagues and still not working

**Comment by CarpetHead** [comment]
Also having the same issue on an lg CX 55

**Comment by Ansem013** [comment]
Same for me, Software Firmware 5.20.15. Hopefully there will be a new version that will support this firmware 🙏

**Comment by FlobertSky** [comment]
Still doesn't work as of March 28th. Has there been any updates on when this could work? Was really looking forward to root my tv what a shame :/

**Comment by theoneandonly-vector** [comment]
please people.. stop spamming.. we all know that it doesn't work anymore on new Software Versions. just posting every model isn't gonna help.

**Comment by loganx82** [comment]
> please people.. stop spamming.. we all know that it doesn't work anymore on new Software Versions. just posting every model isn't gonna help.

Got it! thanks for the advise. 
Now , moving on.. How can we sort out this issue? any suggested method? downgrading? if so, how? 
Perhaps we should wait for updates on this.

Thanks in advance.

**Comment by JasonLandbridge** [comment]
At everyone! I assume you all want to install [The Homebrew Channel](https://github.com/webosbrew/webos-homebrew-channel), which can also be done by [creating a developer account and installing the developer mode app](https://webostv.developer.lge.com/develop/app-test/using-devmode-app/) on the TV. 

Then you can use [DevMode Manager for webOS TV](https://github.com/webosbrew/dev-manager-desktop/tree/master) to connect to your TV from your PC. Once connected, install the Homebrew channel from the application and then you're done!

**Comment by SoraChuu** [comment]
Most people want root access to be able to use Piccap Hyperion which only works with rooted TV. Sadly root still doesn't work on the latest firmware.

Hopefully a new version of RootMyTv is out soon. It shouldn't take long because everything seems to work fine, only at the final step when the TV restarts something goes wrong.

**Comment by sascha005** [comment]
OK, so it is not possible with the use of the DevMode Manager to get Hyperion Working at this time ?

**Comment by RingKing86** [comment]
> OK, so it is not possible with the use of the DevMode Manager to get Hyperion Working at this time ?

Yeah it requires root access to work as of yet.

**Comment by sascha005** [comment]
hm, okay so Fingers crossing for now...

**Comment by cteubs** [comment]
Not sure thre the thread here is version specific, but for what it's worth, the exploit worked on my tv running webOS 6.2.0-35.

**Comment by Reintjuu** [comment]
@JasonLandbridge, a more detailed guide would be appreciated!

@Informatic, is there something we could help with? Do you need more info/logs?

**Comment by StratusFearMe21** [comment]
Can concurr

**Comment by kubo6472** [comment]
Is there any way to downgrade webos?

**Comment by Xploder** [comment]
So my TV just updated aswell (girlfriend confirmed the update prompt... forgot to re-enable blocking).
Really looking forward to an updated exploit to get back hyperion-webos.
Please, if there's something I can help with, hit me up (LG C9, 5.20.15).

**Comment by deniscerri** [comment]
> At everyone! I assume you all want to install [The Homebrew Channel](https://github.com/webosbrew/webos-homebrew-channel), which can also be done by [creating a developer account and installing the developer mode app](https://webostv.developer.lge.com/develop/app-test/using-devmode-app/) on the TV.
> 
> Then you can use [DevMode Manager for webOS TV](https://github.com/webosbrew/dev-manager-desktop/tree/master) to connect to your TV from your PC. Once connected, install the Homebrew channel from the application and then you're done!
> 
> Extra tip: I managed to also install [Ad-free Youtube](https://github.com/webosbrew/youtube-webos) by doing this!
> 
> Ask me if you have any questions and maybe I will write a more detailed guide

Wouldn't homebrew app expire after some time since you need to refresh developer mode, or is it different?

**Comment by pablocp19** [comment]
For those who are interested, there is a way to downgrade, updating to a test version that can be rooted and then downgrade.

1- Install the "Lg Service Remote Control" application on an android device with infrared and press the "in start" button pointing at the screen that opens a password section in which you have to write "0413" and within the configuration in the "System 2" section, an option appears with the name NSU Mode that must be set to "Engineering mode" and exit.
2- Change the location setting of the system  configuration to "Other" in the first section and two hyphens in the second such that "--".
3- Once this is established and restarted, an update to version 30.05.09 should appear, if so, update otherwise NO.
4- Once updated, the rootmy.tv procedure can be done and later it could be downgraded following the existing methods to the version prior to patching.

Once the downgrade is done, you can disable "Engineering mode" and set location to whatever you want.

**Comment by koenvdheuvel** [comment]
> For those who are interested, there is a way to downgrade, updating to a test version that can be rooted and then downgrade.
> 
> 1- Install the "Lg Service Remote Control" application on an android device with infrared and press the "in start" button pointing at the screen that opens a password section in which you have to write "0413" and within the configuration in the "System 2" section, an option appears with the name NSU Mode that must be set to "Engineering mode" and exit. 2- Change the location setting of the system configuration to "Other" in the first section and two hyphens in the second such that "--". 3- Once this is established and restarted, an update to version 30.05.09 should appear, if so, update otherwise NO. 4- Once updated, the rootmy.tv procedure can be done and later it could be downgraded following the existing methods to the version prior to patching.
> 
> Once the downgrade is done, you can disable "Engineering mode" and set location to whatever you want.

Tried it on my OLED55C9PLA but only got an option to update to `05.35.35` in the system software updater.

**Comment by pablocp19** [comment]
> > For those who are interested, there is a way to downgrade, updating to a test version that can be rooted and then downgrade.
> > 1- Install the "Lg Service Remote Control" application on an android device with infrared and press the "in start" button pointing at the screen that opens a password section in which you have to write "0413" and within the configuration in the "System 2" section, an option appears with the name NSU Mode that must be set to "Engineering mode" and exit. 2- Change the location setting of the system configuration to "Other" in the first section and two hyphens in the second such that "--". 3- Once this is established and restarted, an update to version 30.05.09 should appear, if so, update otherwise NO. 4- Once updated, the rootmy.tv procedure can be done and later it could be downgraded following the existing methods to the version prior to patching.
> > Once the downgrade is done, you can disable "Engineering mode" and set location to whatever you want.
> 
> Tried it on my OLED55C9PLA but only got an option to update to `05.35.35` in the system software updater.

I just looked at the compatible models:

 UM7050-UM7090, UM7100-UM7110, UM7300, UM7400-UM7490, UM7500-UM7570, UM7600-UM7660, SM8000, SM8050, SM8100, SM8200 (webOS 4.5)

Sorry

**Comment by Alex-le-vert** [comment]
> For those who are interested, there is a way to downgrade, updating to a test version that can be rooted and then downgrade.
> 
> Compatible models:
> 
> UM7050-UM7090, UM7100-UM7110, UM7300, UM7400-UM7490, UM7500-UM7570, UM7600-UM7660, SM8000, SM8050, SM8100, SM8200 (webOS 4.5)
> 
> 1- Install the "Lg Service Remote Control" application on an android device with infrared and press the "in start" button pointing at the screen that opens a password section in which you have to write "0413" and within the configuration in the "System 2" section, an option appears with the name NSU Mode that must be set to "Engineering mode" and exit. 2- Change the location setting of the system configuration to "Other" in the first section and two hyphens in the second such that "--". 3- Once this is established and restarted, an update to version 30.05.09 should appear, if so, update otherwise NO. 4- Once updated, the rootmy.tv procedure can be done and later it could be downgraded following the existing methods to the version prior to patching.
> 
> Once the downgrade is done, you can disable "Engineering mode" and set location to whatever you want.

Thanks, it's works well for my UM7050 !
How old is this version 30.05.09 ?

**Comment by Mazda77** [comment]
> For those who are interested, there is a way to downgrade, updating to a test version that can be rooted and then downgrade.
> 
> Compatible models:
> 
> UM7050-UM7090, UM7100-UM7110, UM7300, UM7400-UM7490, UM7500-UM7570, UM7600-UM7660, SM8000, SM8050, SM8100, SM8200 (webOS 4.5)
> 
> 1- Install the "Lg Service Remote Control" application on an android device with infrared and press the "in start" button pointing at the screen that opens a password section in which you have to write "0413" and within the configuration in the "System 2" section, an option appears with the name NSU Mode that must be set to "Engineering mode" and exit. 2- Change the location setting of the system configuration to "Other" in the first section and two hyphens in the second such that "--". 3- Once this is established and restarted, an update to version 30.05.09 should appear, if so, update otherwise NO. 4- Once updated, the rootmy.tv procedure can be done and later it could be downgraded following the existing methods to the version prior to patching.
> 
> Once the downgrade is done, you can disable "Engineering mode" and set location to whatever you want.

You forgot to give link to original post on webOS forum: http://webos-forums.ru/post152991.html#p152991 ;)

**Comment by gotwig** [comment]
looking at the commit history, showing latest technical commit in january, i assume the project is officialy dead?

**Comment by Wh1terat** [comment]
Don't give up hope 😉 😉 

OLED65CX running 04.30.61

```
whiterat@xx:[~]$ telnet 10.1.161.192
Trying 10.1.161.192...
Connected to 10.1.161.192.
Escape character is '^]'.

webOS TV 5.3.0 LGwebOSTV

/ # uname -a
Linux LGwebOSTV 4.4.84-219.1.jasper.2 #1 SMP PREEMPT Thu Feb 24 09:02:14 UTC 2022 aarch64 GNU/Linux
/ # id
uid=0 (root) gid=0 (root)
```

**Comment by rotdrop** [comment]
> Don't give up hope wink wink
> 
> OLED65CX running 04.30.61

But wasn't this thread about >= 05.20.15 ???

**Comment by Mazda77** [comment]
> > Don't give up hope wink wink
> > OLED65CX running 04.30.61
> 
> But wasn't this thread about >= 05.20.15 ???

He wrote about new method to get root, including(?) 2019 models with 05.20.15.

**Comment by kubo6472** [comment]
> looking at the commit history, showing latest technical commit in january, i assume the project is officialy dead?

no I rather assume that they are hard at work trying to find vulns in the TVs that are no longer vulnerable now.

**Comment by Informatic** [comment]
FYI we we are in the process of working on v3 chain, which we plan to release in around a month.

**Comment by porroto** [comment]
That's great news. Seriously waiting for an update, came late to root my OLED 55GXPUA WebOS 5.4.1 (Software Version 04.40.10) Couldn't yet

**Comment by nzheltov** [comment]
> FYI we we are in the process of working on v3 chain, which we plan to release in around a month. We **plan** to make it work on all currently released firmware versions (including 2022 TVs), though can't guarantee that yet, since we haven't done enough testing.

Hello! Any plans to support latest OLED G2? Which FW could be supported?

Thanks!

**Comment by DavidBuchanan314** [comment]
> looking at the commit history, showing latest technical commit in january, i assume the project is officialy dead?

We don't develop exploits in the open, because LG would be able to patch them quicker, and we don't want that :)

Rest assured that the project is not dead. but I'm closing this issue for now as a duplicate of #18 - We know it's not working right now, but the next release is comming soon™ and will aim to support as many devices as possible.

**Comment by TimberMeShivers** [comment]
Any word on future updates? Or is 22 the last year of rootmy TV support? :(

**Comment by kubo6472** [comment]
> Any word on future updates? Or is 22 the last year of rootmy TV support? :(

You can root your TV as described up there. I did it that way and it still works.

**Comment by TimberMeShivers** [comment]
I attempted this on my LG B2 and it told me "No vulnerable apps found:("

**Comment by kubo6472** [comment]
@TimberMeShivers use [this](https://gist.github.com/throwaway96/e811b0f7cc2a705a5a476a8dfa45e09f)
